### PR TITLE
librbd: remove consistency group rbd cli and API support

### PR DESCRIPTION
**Pull request by dillaman**
Fixes: http://tracker.ceph.com/issues/18231
Signed-off-by: Jason Dillaman <dillaman@redhat.com>

### CODE REVIEW

File: src/librbd/Group.h [hunk 1]
```diff
@@ -5,6 +5,22 @@
...
+} group_image_status_t;
+
+typedef struct {
+  std::string name;
+  int64_t pool;
+} group_spec_t;
+
 // Consistency groups functions
 int group_create(librados::IoCtx& io_ctx, const char *imgname);
 int group_remove(librados::IoCtx& io_ctx, const char *group_name);
```
Comment (smithfarm) [inline src/librbd/Group.h]: Why are you retaining the function declarations when the corresponding definitions have been dropped?

Comment (dillaman) [inline src/librbd/Group.h]: @smithfarm I am mostly concerned about ensuring the API and cli isn't there -- leaving unused functions is less of a priority to me.

**Comment by liewegas** [comment]
Please hold off on merging this into the kraken branch until we're ready.  Thanks!

**Comment by trociny** [comment]
@dillaman need to update test/cli/rbd/help.t

**Pull request merged**